docs/Reference/Validation-and-Serialization.md

```diff
@@ -22,6 +22,12 @@ specification.
 > it should not be used for initial validation. Accessing databases during
 > validation may lead to Denial of Service attacks. Use
 > [Fastify's hooks](./Hooks.md) like `preHandler` for `async` tasks after validation.
+>
+> ⚠ **Important**: When using custom validators with async `preValidation` hooks,
+> validators **must return** `{error}` objects instead of throwing errors.
+> Throwing errors from custom validators will cause unhandled promise rejections
+> that crash your application when combined with async hooks. See the
+> [custom validator examples](#using-other-validation-libraries) below for the correct pattern.
```

Comment:
```suggestion
>
> When using custom validators with async `preValidation` hooks,
> validators **must return** `{error}` objects instead of throwing errors.
> Throwing errors from custom validators will cause unhandled promise rejections
> that crash the application when combined with async hooks. See the
> [custom validator examples](#using-other-validation-libraries) below for the correct pattern.
```

This is already inside a warning block, so the "important" bit is not needed.
Likewise, use of `you` should be avoided in the formal documentation, see the [documentation style guide](https://fastify.dev/docs/latest/Guides/Style-Guide/#when-to-avoid-the-second-person-you-as-the-pronoun).